src/main/java/io/swagger/codegen/languages/DefaultCodegenConfig.java

```diff
@@ -2518,7 +2518,11 @@ public boolean isDataTypeFile(String dataType) {
                 codegenSecurity.getVendorExtensions().put(CodegenConstants.IS_KEY_IN_QUERY_EXT_NAME, !isKeyInHeader);
 
             } else if (SecurityScheme.Type.HTTP.equals(schemeDefinition.getType())) {
-                codegenSecurity.getVendorExtensions().put(CodegenConstants.IS_BASIC_EXT_NAME, Boolean.TRUE);
+                if ("bearer".equalsIgnoreCase(schemeDefinition.getScheme())) {
+                    codegenSecurity.getVendorExtensions().put(CodegenConstants.IS_BEARER_EXT_NAME, Boolean.TRUE);
+                } else {
```

Comment:
Here you assume that any authentication scheme that is not the bearer authentication scheme is the basic authentication scheme. This is wrong. [There are some others: Digest, HOBA, Mutual.](https://developer.mozilla.org/en-US/docs/Web/HTTP/Authentication#Authentication_schemes). These will fail the first condition in line 2521, since they are not the bearer scheme, but they will also not match the intentions of the second case, namely that they are the basic scheme. You should treat the basic scheme as a special case and then as a first step, all the others in a generic way.

Reply:
thanks for your support @lorenzleutgeb. Mainly i added this code based that we're not supporting the other one, but will follow your idea.